internal/monitors/kubernetes/prometheus/kubernetes.go

```diff
@@ -0,0 +1,85 @@
+// Implements a ConfigInterface that other monitors can extend to gather Prometheus metrics using Kubernetes
+// service account.
+package prometheus
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	k8s "k8s.io/client-go/kubernetes"
+
+	"github.com/signalfx/signalfx-agent/internal/core/common/kubernetes"
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+
+	"github.com/prometheus/common/expfmt"
+	"github.com/signalfx/signalfx-agent/internal/monitors/prometheusexporter"
+)
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"true"`
+
+	// Host of the exporter
+	Host string `yaml:"host" validate:"required"`
```

Comment:
Can you not embed the Prometheus exporter config struct?

Reply:
Prometheus exporter has options like username/password for basic auth that aren't used for Kubernetes APIs (I don't think).